docs/transport.rst

```diff
@@ -0,0 +1,214 @@
+The figure below shows the *Client* transport architecture for the UDP transport.
+
+.. image:: images/client_transport_architecture.svg
+
+Similar to the *Agent* architecture, there is also an interface, ``uxrCommunication``, whose functions are call from the Session Layer.
+That is, each time a ``run_session`` is called, the Session Layer calls to ``send_msg_func`` and ``recv_msg_func`` without worrying about the transport protocol or the platform in use.
+This struct has the following virtual functions:
+
+.. code-block:: c
+
+    bool send_msg_func(void* instance, const uint8_t* buf, size_t len);
+    bool recv_msg_func(void* instance, uint8_t** buf, size_t* len, int timeout);
+    uint8_t comm_error_func(void);
+
+These functions are override by the ``uxrUDPTransport``, which is in charge of two main tasks:
+
+1. Override the communication interface functions. 
+   For example, in the case of the UDP protocol these functions are the followings:
+
+.. code-block:: c
+   
+    bool send_udp_msg(void* instance, const uint8_t* buf, size_t len);
+    bool recv_udp_msg(void* instance, uint8_t** buf, size_t* len, int timeout);
+    uint8_t get_udp_error(void);
+
+2. Offer to the user the initialization and close functions related with the transport protocol.
+   For example, in the case of the UDP protocol these functions are the followings:
+
+.. code-block:: c
+
+    bool uxr_init_udp_transport(uxrUDPTransport* transport, uxrUDPPlatform* platform, const char* ip, uint8_t port);
+    bool uxr_close_udp_transport(uxrUDPTransport* transport);
+
+For each platform there is an implementation of the virtual functions defined in the Transport Layer interface.
+For example, in the case of Linux under UDP transport protocol, the ``uxrUDPPlatform`` implement the following functions:
+
+.. code-block:: c
+
+    bool uxr_init_udp_platform(uxrUDPPlatform* platform, const char* ip, uint16_t port);
+    bool uxr_close_udp_platform(uxrUDPPlatform* platform);
+    size_t uxr_write_udp_data_platform(uxrUDPPlatform* platform, const uint8_t* buf, size_t len, uint8_t* errcode);
+    size_t uxr_read_udp_data_platform(uxrUDPPlatform* platform, uint8_t* buf, size_t len, int timeout, uint8_t* errcode);
+
+Custom Serial Transport
+-----------------------
+*eProsima Micro XRCE-DDS* has a **Custom Serial Transport** with the following features:
+
+* **HDLC Framing**: each serial framing begins with a ``begin_frame`` octet ``(0x7E)`` and the rest of the frame is byte stuffing using the ``space`` octet ``(0x7D)`` following by the original octet exclusive-or with ``0x20``.
+  For example, if the frame contains the octet `0x7E` it is encoded as `0x7D, 0x5E`; and the same for the octet `0x7E` which is encoded as `0x7D, 0x5D`.
+* **CRC Calculation**: serial frames end with a CRC-16 for detecting frame corruption. The CRC-16 is computed using the polynomial ``x^16 + x^12 + x^5 + 1`` after the frame stuffing for each octet of the frame and including the ``begin_frame``.
```

Comment:
I was searching, and exactly, we are using the CCITT version, Maybe we can add this information. But I do not know if it is offtopic.

Reply:
Added suggestion!